plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Simplify.hs

```diff
@@ -74,4 +87,11 @@ simplifyTerm opts builtinSemanticsVariant =
         (Just Refl, Just Refl) -> cse builtinSemanticsVariant
         _                      -> pure
 
+    traceAST ast =
+      case eqT @fun @DefaultFun of
```

Comment:
Why do you need this check here?

Reply:
If I recall correctly, it's to convert to the Agda representation of terms, the Haskell -> Agda intermediate module uses that: https://github.com/IntersectMBO/plutus/blob/4066409bcc355c1f7b63e85271e679621e13be43/plutus-metatheory/src/Untyped.hs#L36.

Reply:
I don't get it. If it was for that function, you'd also need to check equality of `uni` and `DefaultUni`. What breaks if you remove the `eqT` check?